packages/cli/src/constructs/check.ts

```diff
@@ -16,90 +16,215 @@ import { ConfigDefaultsGetter, makeConfigDefaultsGetter } from './check-config'
 import { Diagnostics } from './diagnostics'
 import { validateDeprecatedDoubleCheck } from './internal/common-diagnostics'
 
+/**
```

Comment:
@sorccu Hmmm... I'm unsure of including these docs because people would never interact with `CheckProps` or am I wrong?

Reply:
Not usually no, but I think the example is valid - if you wanted to have maintain a common set of properties for your checks, this would be the way to do it in a type safe way. You'd then use it like this:

```
new ApiCheck('foo', {
  ...commonCheckProps,
  name: 'Foo',
})
```

The example could be expanded to elaborate on this use case though.

Reply:
I'll delete the JSDoc here.